src/main/java/login/ui/LogInView.java

```diff
@@ -0,0 +1,110 @@
+package login.ui;
+
+import login.interface_adapters.LogInController;
+import login.interface_adapters.LogInViewModel;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class LogInView extends JPanel implements ActionListener {
+
+    private final LogInViewModel viewModel;
+
+    private final LogInController controller;
+
+    private final JTextArea enterEmail = new JTextArea("");
+    private final JTextArea enterPassword = new JTextArea("");
+
+
+    public LogInView(LogInController controller, LogInViewModel viewModel){
+        this.controller = controller;
+        this.viewModel = viewModel;
+
+
+        this.setLayout(new GridBagLayout());
+        GridBagConstraints constraints = new GridBagConstraints();
+
+        JLabel textLabel = new JLabel("Log In");
+        constraints.fill = GridBagConstraints.HORIZONTAL;
+        constraints.weightx = 0.5;
+        constraints.gridx = 5;
+        constraints.gridy = 6;
+        this.add(textLabel, constraints);
+
+        JLabel email = new JLabel("Email");
+        constraints.fill = GridBagConstraints.HORIZONTAL;
+        constraints.weightx = 0.5;
+        constraints.gridx = 3;
+        constraints.gridy = 8;
+        this.add(email, constraints);
+
+
+        constraints.fill = GridBagConstraints.HORIZONTAL;
+        constraints.weightx = 0.5;
+        constraints.gridx = 3;
+        constraints.gridy = 10;
+        this.add(enterEmail, constraints);
+
+        constraints.fill = GridBagConstraints.HORIZONTAL;
+        constraints.weightx = 0.5;
+        constraints.gridx = 3;
+        constraints.gridy = 12;
+        this.add(enterPassword, constraints);
+
+        JButton logInButton = new JButton("Log In");
+        logInButton.addActionListener(this);
+        constraints.fill = GridBagConstraints.HORIZONTAL;
+        constraints.weightx = 0.5;
+        constraints.gridx = 3;
+        constraints.gridy = 14;
+        this.add(logInButton, constraints);
+
+        this.setVisible(true);
+    }
+
+
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        String emailInput = enterEmail.getText();
+        String passwordInput = enterPassword.getText();
+
+        enterEmail.setText(" ");
+        enterPassword.setText(" ");
+
+        controller.logInInitializer(emailInput, passwordInput);
+        if (viewModel.getLogInSuccess()){
+            // call main view
```

Comment:
I think the best way to access main view would be to have it as an observable for the view, and tell the observables to update themselves. See Richards or my use case for an example of that.